File: location/src/main/java/com/senjuid/location/LocationPluginOptions.java
package com.senjuid.location;

public class LocationPluginOptions {
    public String data;
    public String message1;
    public String message2;

    private LocationPluginOptions(Builder builder) {
        this.data = builder.data;
        this.message1 = builder.message1;
        this.message2 = builder.message2;
    }

    public static class Builder {
        private String data;
        private String message1;
        private String message2;

        public Builder setData(String data) {
            this.data = data;
            return this;
        }

        public Builder setMessage(String message1, String message2) {
            this.message1 = message1;
            this.message2 = message2;
            return this;
        }

        public LocationPluginOptions build() {
            return new LocationPluginOptions(this);
        }
    }
}

File: location/src/main/java/com/senjuid/location/LocationPluginOptions.kt
package com.senjuid.location

class LocationPluginOptions private constructor(
        var data: String?,
        var message1: String?,
        var message2: String?
) {
    data class Builder(
            var data: String? = null,
            var message1: String? = null,
            var message2: String? = null
    ) {
        fun setMessage(message1: String?, message2: String?) = apply { this.message1 = message1 }.apply { this.message2 = message2 }
        fun setData(data: String) = apply { this.data = data }
        fun build() = LocationPluginOptions(data, message1, message2)
    }
}